Source/Core/InputCommon/ControllerInterface/GameController/GameControllerDevice.mm

```diff
@@ -0,0 +1,109 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/ControllerInterface/GameController/GameControllerDevice.h"
+
+#include "Common/Logging/Log.h"
+#include "Common/MathUtil.h"
+
+namespace ciface::GameController
+{
+using MathUtil::GRAVITY_ACCELERATION; // 9.80665
+
+Controller::Controller(const GCController* controller) : m_controller(controller)
+{
+  std::vector<std::pair<std::string, GCControllerButtonInput*>> pairs;
+    
+  for (GCControllerButtonInput* item in controller.physicalInputProfile.allButtons)
+  {
+    pairs.push_back(std::make_pair(std::string([item.aliases.allObjects[0] UTF8String]), item));
+  }
+    
+  sort(pairs.begin(), pairs.end(), [=](std::pair<std::string, GCControllerButtonInput*>& a, std::pair<std::string, GCControllerButtonInput*>& b)
+  {
+    return a.first.compare(b.first)<0;
+  }
+  );
+    
+  for (size_t i = 0; i < pairs.size(); ++i) {
+    INFO_LOG_FMT(SERIALINTERFACE, "Found button '{}'", pairs[i].first);
+    AddInput(new Button(pairs[i].second));
+  }
+    
+  if (controller.motion)
```

Comment:
```suggestion
  if (!controller.motion)
    return;
```

Reply:
Personal preference I guess, but I'd use m_controller instead of controller inside this function.